core/src/banking_simulation.rs

```diff
@@ -0,0 +1,616 @@
+                        );
+                    }
+
+                    if exit.load(Ordering::Relaxed) {
+                        break;
+                    }
+                }
+                info!(
+                    "terminating to send...: non_vote: {} ({}), tpu_vote: {} ({}), gossip_vote: {} ({})",
+                    non_vote_count,
+                    non_vote_tx_count,
+                    tpu_vote_count,
+                    tpu_vote_tx_count,
+                    gossip_vote_count,
+                    gossip_vote_tx_count
+                );
+                // hold these senders in join_handle to control banking stage termination!
+                (non_vote_sender, tpu_vote_sender, gossip_vote_sender)
+            }
+        })?;
+
+        sleep(warmup_duration);
+        info!("warmup done!");
+
+        loop {
+            if poh_recorder.read().unwrap().bank().is_none() {
+                let next_leader_slot = leader_schedule_cache.next_leader_slot(
+                    &simulated_leader,
+                    bank.slot(),
+                    &bank,
+                    Some(&blockstore),
+                    GRACE_TICKS_FACTOR * MAX_GRACE_SLOTS,
+                );
+                debug!("{next_leader_slot:?}");
+                poh_recorder
+                    .write()
+                    .unwrap()
+                    .reset(bank.clone_without_scheduler(), next_leader_slot);
+                info!("Bank::new_from_parent()!");
+
+                let old_slot = bank.slot();
+                if let Some(event_time) =
+                    self.banking_trace_events.freeze_time_by_slot.get(&old_slot)
+                {
+                    if log_enabled!(log::Level::Info) {
+                        let current_simulation_time = SystemTime::now();
+                        let elapsed_simulation_time = current_simulation_time
+                            .duration_since(base_simulation_time)
+                            .unwrap();
+                        let elapsed_event_time =
+                            event_time.duration_since(base_event_time).unwrap();
+                        info!(
+                            "jitter(parent_slot: {}): {}{:?} (sim: {:?} event: {:?})",
+                            old_slot,
+                            if elapsed_simulation_time > elapsed_event_time {
+                                "+"
+                            } else {
+                                "-"
+                            },
+                            if elapsed_simulation_time > elapsed_event_time {
+                                elapsed_simulation_time - elapsed_event_time
+                            } else {
+                                elapsed_event_time - elapsed_simulation_time
+                            },
+                            elapsed_simulation_time,
+                            elapsed_event_time,
+                        );
```

Comment:
this particular log output are like this:

```
$ grep -E 'jitter' simulate-mb.2024y08m26d10h44m43s933116486ns
[2024-08-26T10:58:57.989324285Z INFO  solana_core::banking_simulation] jitter(parent_slot: 282254383): +360.27µs (sim: 12.00036027s event: 12s)
[2024-08-26T10:58:58.344810251Z INFO  solana_core::banking_simulation] jitter(parent_slot: 282254384): -19.615829ms (sim: 12.355846357s event: 12.375462186s)
[2024-08-26T10:58:58.695797971Z INFO  solana_core::banking_simulation] jitter(parent_slot: 282254385): -71.903503ms (sim: 12.706834067s event: 12.77873757s)
[2024-08-26T10:58:59.047995512Z INFO  solana_core::banking_simulation] jitter(parent_slot: 282254386): -135.656223ms (sim: 13.059031477s event: 13.1946877s)
```

in short, poh in sim is rather more timely than the actual traced poh recordings. maybe this is due to much reduced sysload.